serve_event.go

```diff
@@ -0,0 +1,117 @@
+package wiremock
+
+type ServeEvents struct {
+	ServeEvents []ServeEvent `json:"serveEvents,omitempty"`
+}
+
+type ServeEvent struct {
+	RequestJournalDisabled bool             `json:"requestJournalDisabled,omitempty"`
+	Meta                   Meta             `json:"meta,omitempty"`
+	Requests               []RequestElement `json:"requests,omitempty"`
+
+	Request            ServeEventRequest       `json:"request,omitempty"`
+	StubMapping        StubMapping             `json:"stubMapping,omitempty"`
+	WasMatched         bool                    `json:"wasMatched,omitempty"`
+	Response           PurpleResponse          `json:"response,omitempty"`
+	Timing             Timing                  `json:"timing,omitempty"`
+	ID                 string                  `json:"id,omitempty"`
+	ResponseDefinition ResponseDefinitionClass `json:"responseDefinition,omitempty"`
+}
+
+type Meta struct {
+	Total int64 `json:"total,omitempty"`
+}
+
+type RequestElement struct {
+	Request            ServeEventRequest       `json:"request,omitempty"`
+	StubMapping        StubMapping             `json:"stubMapping,omitempty"`
+	WasMatched         bool                    `json:"wasMatched,omitempty"`
+	Response           PurpleResponse          `json:"response,omitempty"`
+	Timing             Timing                  `json:"timing,omitempty"`
+	ID                 string                  `json:"id,omitempty"`
+	ResponseDefinition ResponseDefinitionClass `json:"responseDefinition,omitempty"`
+}
+
+type ServeEventRequest struct {
+	Headers             RequestHeaders `json:"headers,omitempty"`
+	Method              string         `json:"method,omitempty"`
+	Scheme              string         `json:"scheme,omitempty"`
+	BrowserProxyRequest bool           `json:"browserProxyRequest,omitempty"`
+	QueryParams         Cookies        `json:"queryParams,omitempty"`
+	AbsoluteURL         string         `json:"absoluteUrl,omitempty"`
+	Body                string         `json:"body,omitempty"`
+	LoggedDate          int64          `json:"loggedDate,omitempty"`
+	URL                 string         `json:"url,omitempty"`
+	Cookies             Cookies        `json:"cookies,omitempty"`
+	Protocol            string         `json:"protocol,omitempty"`
+	Port                int64          `json:"port,omitempty"`
+	LoggedDateString    string         `json:"loggedDateString,omitempty"`
+	ClientIP            string         `json:"clientIp,omitempty"`
+	Host                string         `json:"host,omitempty"`
+	BodyAsBase64        string         `json:"bodyAsBase64,omitempty"`
+}
+
+type Cookies struct {
+}
```

Comment:
This can be a generic map
```suggestion
type Cookies map[string]string
```